heat_pump_adoption_modelling/analysis/MCS/plot_mcs.py

```diff
@@ -0,0 +1,549 @@
+from heat_pump_adoption_modelling.pipeline.preprocessing.mcs_epc_joining import (
+    join_mcs_epc_data,
+)
+
+# Load data for plotting
+
+
+def plottable_data(file_path=PROJECT_DIR / "outputs/mcs_epc.csv"):
+    """Load merged MCS/EPC data in a suitable form for plotting
+    (with additional variables: "built_post_1950" indicating
+    whether or not the property was built after 1950, and
+    "inflated_cost" indicating inflation-adjusted installation cost).
+
+    Parameters
+    ----------
+    file_path : str
+        Path to saved merged data.
+        If none, data is loaded from scratch.
+
+    Return
+    ----------
+    merged : pd.DataFrame
+        Merged MCS-EPC data with added variables.
+    """
+
+    if file_path is None:
+        merged = join_mcs_epc_data()
+    else:
+        merged = pd.read_csv(file_path)
+
+    # Add inflated cost variable
+    inflation = load_inflation()
+    merged = merged.merge(inflation, how="left", on="year")
+    merged["inflated_cost"] = merged["cost"] * merged["multiplier"]
+
+    # Add indicator variable for properties built after 1950
+    ages_dict = {
+        "England and Wales: before 1900": False,
+        "1900-1929": False,
+        "1930-1949": False,
+        "1950-1966": True,
+        "1965-1975": True,
+        "1976-1983": True,
+        "1983-1991": True,
+        "1991-1998": True,
+        "1996-2002": True,
+        "2003-2007": True,
+        "2007 onwards": True,
+        "unknown": pd.NA,
+        np.nan: pd.NA,
+        None: pd.NA,
+    }
+
+    merged["built_post_1950"] = merged["CONSTRUCTION_AGE_BAND"].apply(
+        lambda x: ages_dict[x]
+    )
+
+    return merged
+
+
+# Define functions to assist with "archetype" boxplots
+
+
+def bool_not(series):
+    """Given a series of bools and NAs, return a series
+    with NAs in the same places but bools inverted.
+
+    Parameters
+    ----------
+    series : pd.Series
+        A series of bools and NAs.
+
+    Return
+    ----------
+    result : pd.Series
+        A series with NAs in the same places and bools inverted.
+    """
+
+    # Initialise with a series of NAs
+    result = pd.Series(np.repeat(pd.NA, len(series)))
+
+    result.loc[series == True] = False
+    result.loc[series == False] = True
+
+    return result
+
+
+def generate_archetype_dict(df):
+    """Given a dataframe, produce a dictionary of archetype labels
+    to readable names and logical vectors indicating the rows of
+    the dataframe that conform to that archetype.
+    Archetypes are defined in the top table of sheet "DAv2" here:
+    https://docs.google.com/spreadsheets/d/1HkUYneexFXDTgMyJs_UaUbBmXGUum6I1K4L2GP2rEjQ/edit?usp=sharing
+
+    Parameters
+    ----------
+    df : pd.DataFrame
+        Dataframe with "PROPERTY_TYPE", "BUILT_FORM" and "built_post_1950"
+        variables.
+
+    Return
+    ----------
+    result : dict
+        A dictionary from archetype numbers to names and logical vectors.
+    """
+
+    archetype_dict = {
```

Comment:
Why do you need a dict for this? I don't see you using something like _archetype_dict["archetype_1"]_ anywhere. 

Further down you end up iterating over the values in this dict, which would be easier with a list, especially since dicts are even unordered.

Why not save this as a tuple list?

```
archetypes = [("Post-1950 flats", (df.PROPERTY_TYPE == "Flat") & (df.built_post_1950)),
   "Pre-1950 flats", (...) 
... ]
```

I'll explain further down how to use it.

Reply:
This structure is much more sensible :) thanks!